interpreter/k8s/src/test/scala/io/buoyant/transformer/k8s/DaemonSetTransformerTest.scala

```diff
@@ -134,4 +140,161 @@ class DaemonSetTransformerTest extends FunSuite {
       assert(bound.addr.sample == Addr.Neg)
     }
   }
+
+  private[this] def parse(yaml: String): DaemonSetTransformerConfig =
+    Parser.objectMapper(yaml, Iterable(Seq(DaemonSetTransformerInitializer)))
+      .readValue[TransformerConfig](yaml)
+      .asInstanceOf[DaemonSetTransformerConfig]
+
+  val yaml_hostNetwork_false =
+    s"""|kind: io.l5d.k8s.daemonset
+        |k8sHost: "foo"
+        |k8sPort: 8888
+        |namespace: "test-namespace"
+        |service: "test-service"
+        |port: test-port
+        |hostNetwork: false
+        |""".stripMargin
+
+  val yaml_hostNetwork_true =
+    s"""|kind: io.l5d.k8s.daemonset
+        |k8sHost: "foo"
+        |k8sPort: 9999
+        |namespace: "test-namespace2"
+        |service: "test-service2"
+        |port: test-port2
+        |hostNetwork: true
+        |""".stripMargin
+
+  test("daemonset transformer has admin handler with expected url path") {
+    val config = parse(yaml_hostNetwork_false)
+    inside(config) {
+      case DaemonSetTransformerConfig(k8sHost, k8sPort, namespace, service, port, hostNetwork) =>
+        assert(k8sHost.contains("foo"))
+        assert(k8sPort.contains(Port(8888)))
+        assert(namespace.contains("test-namespace"))
+        assert(service.contains("test-service"))
+        assert(port.contains("test-port"))
+        assert(hostNetwork.contains(false))
+    }
+
+    val transfomer: GatewayTransformer = config.mk(Stack.Params.empty).asInstanceOf[GatewayTransformer]
+    transfomer.adminHandlers.headOption match {
+      case Some(h) =>
+        assert(h.url == "/namer_state/io.l5d.k8s.daemonset/test-namespace/test-port/test-service.json")
+      case None =>
+        fail("DaemonSetTransformer does not have a watch state handler configured")
+    }
+  }
+
+  test("when wrapping a interpreter, the new transformed interpreter implements trait WithNameTreeTransformer and is able to stack transformers") {
+    val transformer = parse(yaml_hostNetwork_false).mk(Stack.Params.empty)
+
+    val interpreter = new NameInterpreter {
+      override def bind(
+                         dtab: Dtab,
+                         path: Path
+                       ): Activity[NameTree[Bound]] = {
```

Comment:
Wow, is that intellij's formatter. If so, its quite strange

Reply:
Not sure, I've copied it from another test, but I'll change the formatting.